avm/ptn/authorization/pim-role-assignment/modules/management-group.bicep

```diff
@@ -0,0 +1,161 @@
+  'Resource Policy Contributor': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    '36243c78-bf99-498c-9df9-86d9f8d28608'
+  )
+  'Role Based Access Control Administrator': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    'f58310d9-a9f6-439a-9e8d-f62e7b41a168'
+  )
+  'User Access Administrator': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    '18d7d88d-d35e-4fb5-a5c3-7773c20a72d9'
+  )
+  'Management Group Reader': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    'ac63b705-f282-497d-ac71-919bf39d939d'
+  )
+}
+
+var roleDefinitionIdVar = (builtInRoleNames[?roleDefinitionIdOrName] ?? roleDefinitionIdOrName)
+
+resource pimEligibleRoleAssignment 'Microsoft.Authorization/roleEligibilityScheduleRequests@2022-04-01-preview' = if (pimRoleAssignmentType.roleAssignmentType == 'Eligible') {
+  name: guid(deployment().name, managementGroupId, roleDefinitionIdVar, principalId)
+  properties: {
+    principalId: principalId
+    roleDefinitionId: roleDefinitionIdVar
+    requestType: requestType
+    condition: !empty(condition) ? condition : null
+    conditionVersion: !empty(conditionVersion) && !empty(condition) ? conditionVersion : null
+    justification: justification
+    targetRoleEligibilityScheduleId: targetRoleEligibilityScheduleId
+    targetRoleEligibilityScheduleInstanceId: targetRoleEligibilityScheduleInstanceId
+    ticketInfo: ticketInfo
+    scheduleInfo: pimRoleAssignmentType.scheduleInfo.durationType == 'NoExpiration'
+      ? {
+          expiration: {
+            type: 'NoExpiration'
+          }
+        }
+      : pimRoleAssignmentType.scheduleInfo.durationType == 'AfterDuration'
+          ? {
+              expiration: {
+                type: 'AfterDuration'
+                duration: pimRoleAssignmentType.scheduleInfo.duration
+              }
+              startDateTime: pimRoleAssignmentType.scheduleInfo.startTime
+            }
+          : pimRoleAssignmentType.scheduleInfo.durationType == 'AfterDateTime'
+              ? {
+                  expiration: {
+                    type: 'AfterDateTime'
+                    endDateTime: pimRoleAssignmentType.scheduleInfo.endDateTime
+                  }
+                  startDateTime: pimRoleAssignmentType.scheduleInfo.startTime
+                }
+              : null
+  }
+}
+
+resource pimActiveRoleAssignment 'Microsoft.Authorization/roleAssignmentScheduleRequests@2022-04-01-preview' = if (pimRoleAssignmentType.roleAssignmentType == 'Active') {
+  name: guid(deployment().name, managementGroupId, roleDefinitionIdVar, principalId)
+  properties: {
+    principalId: principalId
+    roleDefinitionId: roleDefinitionIdVar
+    requestType: requestType
+    condition: !empty(condition) ? condition : null
+    conditionVersion: !empty(conditionVersion) && !empty(condition) ? conditionVersion : null
```

Comment:
Should be handled as per the other comments for `condition` & `conditionVersion`. Same applies to all other references of the two.

Reply:
Resolved